warehouse/query-core/src/main/java/datawave/query/tables/ShardQueryLogic.java

```diff
@@ -904,6 +868,12 @@ public QueryLogicTransformer getTransformer(Query settings) {
             transformer.setProjectFields(config.getProjectFields());
             transformer.setBlacklistedFields(config.getBlacklistedFields());
         }
+        if (settings.findParameter(QueryParameters.UNIQUE_FIELDS).getParameterValue().length() > 0) {
+            transformer.addTransform(new UniqueTransform(this, config.getUniqueFields()));
+        }
+        if (settings.findParameter(QueryParameters.GROUP_FIELDS).getParameterValue().length() > 0) {
+            transformer.addTransform(new GroupingTransform(this, config.getGroupFields()));
+        }
```

Comment:
does the order here matter? Is it possible that unique shouldn't be applied until after the group transform?

Reply:
I think the order does matter.  the GroupingTransform always returns null on calls to apply.  In the DocumentTransformer, it stops applying the transformers to the document entry if a null is returned by the apply method.  So, if the grouping transform runs first, the unique transform won't run, right?

Reply:
However, it probably doesn't make sense to ever run the grouping followed by unique.  The way it is now, assuming they are running against the same fields, group by will return a count of 1 for every entry returned by unique.  If the grouping is run on a completely different set of fields, I would expect nothing to be returned.  If it's run on a subset of the fields used for unique, I would expect it to group and count accordingly.

If you group first, and then run unique, the unique transform will do nothing if the fields are the same between the transforms.  If the fields are completely different for unique, I would expect nothing to be returned.  If it's run on a subset of the fields used for unique, I guess it would return all of the unique counts?  Is that useful?

Reply:
It would be nice to test all the different permutations of unique and group by (i.e. unique on set of fields then group by subset of fields, or unique on set of fields then group by a totally different field)